test/eventpattern.test.ts

```diff
@@ -0,0 +1,797 @@
+            (event.detail.num > 300 || event.detail.num < 200)
+        ),
+        {
+          detail: {
+            num: [{ number: [">", 300] }, { number: ["<", 200] }],
+          },
+        }
+      );
+    });
+
+    test("numeric range or and AND part reduced both valid", () => {
+      ebEventPatternTestCase(
+        reflect<EventPredicateFunction<TestEvent>>(
+          (event) =>
+            (event.detail.num > 300 || event.detail.num < 200) &&
+            (event.detail.num > 250 || event.detail.num < 100)
+        ),
+        {
+          detail: {
+            num: [{ number: [">", 300] }, { number: ["<", 100] }],
+          },
+        }
+      );
+    });
+
+    test("numeric range multiple distinct segments", () => {
+      ebEventPatternTestCase(
+        reflect<EventPredicateFunction<TestEvent>>(
+          (event) =>
+            (event.detail.num > 300 && event.detail.num < 400) ||
+            (event.detail.num > 0 && event.detail.num < 200) ||
+            (event.detail.num > -100 && event.detail.num < -50)
+        ),
+        {
+          detail: {
+            num: [
+              { number: [">", 300, "<", 400] },
+              { number: [">", 0, "<", 200] },
+              { number: [">", -100, "<", -50] },
+            ],
+          },
+        }
+      );
+    });
+
+    test("numeric range multiple distinct segments overlapped", () => {
+      ebEventPatternTestCase(
+        reflect<EventPredicateFunction<TestEvent>>(
+          (event) =>
+            (event.detail.num > 300 && event.detail.num < 400) ||
+            (event.detail.num > 0 && event.detail.num < 200) ||
+            (event.detail.num > -100 && event.detail.num < -50) ||
+            event.detail.num > -200
+        ),
+        {
+          detail: {
+            num: [{ number: [">", -200] }],
+          },
+        }
+      );
+    });
+
+    test("numeric range multiple distinct segments merged", () => {
+      ebEventPatternTestCase(
+        reflect<EventPredicateFunction<TestEvent>>(
+          (event) =>
+            (event.detail.num > 300 && event.detail.num < 400) ||
+            (event.detail.num > 0 && event.detail.num < 200) ||
+            (event.detail.num > -100 && event.detail.num < -50) ||
+            (event.detail.num > -200 && event.detail.num < 200)
+        ),
+        {
+          detail: {
+            num: [
+              { number: [">", -200, "<", 200] },
+              { number: [">", 300, "<", 400] },
+            ],
+          },
+        }
+      );
+    });
+
+    // Note: another option would be to just drop the invalid range.
+    test("numeric range or and AND error", () => {
+      ebEventPatternTestCaseError(
+        reflect<EventPredicateFunction<TestEvent>>(
+          (event) =>
+            (event.detail.num > 300 || event.detail.num < 200) &&
+            event.detail.num > 400
+        )
+      );
+    });
+
+    test("numeric range nil range error upper", () => {
+      ebEventPatternTestCaseError(
```

Comment:
Can we also include the error message in the test case here? I would like to know what error message is reported to the user in this case.

Reply:
Yeah, I was planning to get back and set them all once things settled down.